recipes/patchelf/all/conanfile.py

```diff
@@ -23,11 +28,16 @@ def build_requirements(self):
         self.build_requires("libtool/2.4.6")
 
     def validate(self):
-        if not tools.is_apple_os(self.settings.os) and self.settings.os not in ("FreeBSD", "Linux"):
+        if not is_apple_os(self) and self.settings.os not in ("FreeBSD", "Linux"):
             raise ConanInvalidConfiguration("PatchELF is only available for GNU-like operating systems (e.g. Linux)")
 
+        if Version(self.version) >= "0.15" \
```

Comment:
When a C++ standard is required, follow the template: https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/cmake_package/all/conanfile.py#L38

Reply:
I'm probably missing something, but the check logic here doesn't seem to work: https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/cmake_package/all/conanfile.py#L78-L83

```
ERROR: patchelf/0.15: Error in validate() method, line 46
        if self.info.settings.compiler.cppstd:
        AttributeError: 'NoneType' object has no attribute 'cppstd'
```

Similarly, the minimum_version check below never ends up actually happening since it tries to look up `None` in the map rather than the compiler name.

I'm testing with a fresh Conan 1.52.0 install in an Ubuntu 20.04 container.

Reply:
@sfackler it occurs because compiler has been removed from the package ID. To fix your situation, you need to use `get_safe`:

    self.info.settings.get_safe("compiler.cppstd")

Reply:
I pushed up changes that just leave the compiler in the package_id for now.